### PR TITLE
✨ Angular velocity sensor interface

### PR DESCRIPTION
**Pull request by Coreyboy1820**
Added the interface and the tests for inheriting, creation, and errors.

### CODE REVIEW

File: demos/CMakeLists.txt [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
```
Comment (kammce) [inline demos/CMakeLists.txt]: Fix whatever is happening with the license area.

File: demos/CMakeLists.txt [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-message(STATUS "Generating Demo for \"${PROJECT_NAME}")
-add_executable(${PROJECT_NAME} main.cpp)
-target_include_directories(${PROJECT_NAME} PUBLIC .)
-target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_20)
-target_link_libraries(${PROJECT_NAME} PRIVATE libhal::libhal)
+      message(STATUS "Generating Demo for \"${PROJECT_NAME}")
+        add_executable(${ PROJECT_NAME } main.cpp)
+          target_include_directories(${ PROJECT_NAME } PUBLIC.)
+            target_compile_features(${ PROJECT_NAME } PRIVATE cxx_std_20)
+              target_link_libraries(${ PROJECT_NAME } PRIVATE libhal::libhal)
```
Comment (kammce) [inline demos/CMakeLists.txt]: Not sure where the indents and strange formatting is coming from, but can you fix this?

**Review by kammce — CHANGES_REQUESTED** [review]
Besides the formatting, this looks great. I will approve this PR once once the formatting is fixed.

File: demos/CMakeLists.txt (hunk 1 above)
Comment (Coreyboy1820) [reply demos/CMakeLists.txt]: Yeah, I think my clang-format was running on non-c++ files, ill fix my bash folder workaround to only format c++ files.

File: .clang-format [hunk 3]
```diff
@@ -6,4 +6,4 @@ AlwaysBreakAfterDefinitionReturnType: None
 AllowShortFunctionsOnASingleLine: None
 FixNamespaceComments: true
 SpacesBeforeTrailingComments: 2
-ColumnLimit: 80
+ColumnLimit: 80
```
Comment (kammce) [inline .clang-format]: Add newline back

**Review by kammce — CHANGES_REQUESTED** [review]
The files still have formatting issues. See the python program for an example of what happens to the first line.

File: .clang-format (hunk 3 above)
Comment (Coreyboy1820) [reply .clang-format]: Odd, I changed this already but will do it again.

**Comment by Coreyboy1820** [comment]
> The files still have formatting issues. See the python program for an example of what happens to the first line.

Apologies, should all be fixed now.

**Review by kammce — APPROVED** [review]

**Pull request merged**